scripts/west_commands/runners/pyocd.py

```diff
@@ -131,16 +140,29 @@ def flash(self, **kwargs):
                 'Cannot flash; no hex ({}) or bin ({}) files found. '.format(
                     self.hex_name, self.bin_name))
 
-        cmd = ([self.pyocd] +
-               ['flash'] +
-               ['-e', 'sector'] +
-               self.flash_addr_args +
-               self.daparg_args +
-               self.target_args +
-               self.board_args +
-               self.frequency_args +
-               self.flash_extra +
-               [fname])
+        if self.tool_opt_args != []:
```

Comment:
It doesn't look like it got fixed; did you forget to push? There's still a lot of copy pasting that isn't necessary if the tool_opt_args attribute is always a list.

Adding an empty list onto an existing list makes no changes:

```
>>> l = [1,2,3]
>>> l + []
[1, 2, 3]
```

So there's no need for `if self.tool_opt_args != []` here or elsewhere in the file, because if `self.tool_opt_args == []`, then adding it onto the list being saved into `cmd` will have no effect.

Reply:
Sorry for this, I fixed， Thanks！